### PR TITLE
test: Add nightly repeated unit test

### PR DESCRIPTION
**Pull request by royjit**
## Description
Adds the unit test to run repeated iteration nightly.
Sample test run - https://github.com/aws-amplify/amplify-swift/actions/runs/3899004585/jobs/6658264423

## General Checklist

- [ ] ~Added new tests to cover change, if needed~
- [x] Build succeeds with all target using Swift Package Manager
- [x] All unit tests pass
- [x] All integration tests pass
- [ ] ~Security oriented best practices and standards are followed (e.g. using input sanitization, principle of least privilege, etc)~
- [ ] ~Documentation update for the change if required~
- [x] PR title conforms to conventional commit style
- [ ] ~If breaking change, documentation/changelog update with migration instructions~

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by jcjimenez — APPROVED** [review]
LGTM

File: .github/workflows/nightly_repeated_unittest.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+name: Amplify Nightly Unit Test
```
Comment (thisisabhash) [inline .github/workflows/nightly_repeated_unittest.yml:1]: suggestion: renaming to Amplify Nightly Repeated Unit Test Run

File: .github/workflows/nightly_repeated_unittest.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+jobs:
+  unit_test_ios:
+    runs-on: macos-12
+    strategy:
+      matrix:
+        scheme: [Amplify, AWSPluginsCore, AWSPinpointAnalyticsPlugin, AWSAPIPlugin, AWSCognitoAuthPlugin, AWSDataStorePlugin, AWSLocationGeoPlugin, AWSS3StoragePlugin]
+    steps:
+      - uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b
+
+      - name: Run unit test iOS
```
Comment (thisisabhash) [inline .github/workflows/nightly_repeated_unittest.yml:23]: suggestion: renaming to Run Repeated unit test iOS

File: .github/workflows/nightly_repeated_unittest.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+  unit_test_macos:
+    runs-on: macos-12
+    strategy:
+      matrix:
+        scheme: [Amplify, AWSPluginsCore, AWSPinpointAnalyticsPlugin, AWSAPIPlugin, AWSCognitoAuthPlugin, AWSDataStorePlugin, AWSLocationGeoPlugin, AWSS3StoragePlugin]
+    steps:
+      - uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b
+
+      - name: Run unit test macOS
```
Comment (thisisabhash) [inline .github/workflows/nightly_repeated_unittest.yml:38]: suggestion: renaming to Run Repeated unit test macOS

File: .github/workflows/nightly_repeated_unittest.yml (hunk 1 above)
Comment (royjit) [reply .github/workflows/nightly_repeated_unittest.yml]: Changed

File: .github/workflows/nightly_repeated_unittest.yml (hunk 2 above)
Comment (royjit) [reply .github/workflows/nightly_repeated_unittest.yml]: Changed

File: .github/workflows/nightly_repeated_unittest.yml (hunk 3 above)
Comment (royjit) [reply .github/workflows/nightly_repeated_unittest.yml]: Changed

**Review by thisisabhash — APPROVED** [review]

**Pull request merged**